adapter-base-quarkus/src/main/java/org/eclipse/hono/adapter/quarkus/KafkaRuntimeConfigProducer.java

```diff
@@ -0,0 +1,114 @@
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+import javax.inject.Named;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Parses Hono's Kafka configuration properties and provides them as beans.
+ * <p>
+ * This overwrites the behavior of {@link io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer} from Quarkus'
+ * Kafka extension. In addition, the dependency provides some patches that are required to use the Kafka clients in
+ * native images with GraalVM.
+ */
+@Dependent
+public class KafkaRuntimeConfigProducer {
+
+    /**
+     * The prefix for Hono's common Kafka configuration.
+     */
+    public static final String COMMON_CONFIG_PREFIX = "hono.kafka.commonClientConfig";
+
+    /**
+     * The prefix for Hono's Kafka producer configuration.
+     */
+    public static final String PRODUCER_CONFIG_PREFIX = "hono.kafka.producerConfig";
+
+    /**
+     * The prefix for Hono's Kafka consumer configuration.
+     */
+    public static final String CONSUMER_CONFIG_PREFIX = "hono.kafka.consumerConfig";
+
+    /**
+     * The prefix for Hono's Kafka admin client configuration.
+     */
+    public static final String ADMIN_CLIENT_CONFIG_PREFIX = "hono.kafka.adminClientConfig";
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaRuntimeConfigProducer.class);
+
+    /**
+     * Exposes Hono's Kafka configuration properties as a bean.
+     * <p>
+     * The configuration properties are derived from the provided configuration with the prefixes
+     * {@link #COMMON_CONFIG_PREFIX}, {@link #ADMIN_CLIENT_CONFIG_PREFIX}, {@link #CONSUMER_CONFIG_PREFIX}, and
+     * {@link #PRODUCER_CONFIG_PREFIX}.
+     *
+     * @return The configuration properties.
+     */
+    @Produces
+    @ApplicationScoped
+    @Named("default-kafka-broker")
+    public Map<String, Object> createKafkaRuntimeConfig() {
+        final Config config = ConfigProvider.getConfig();
+
+        final Map<String, Object> properties = new HashMap<>();
+        properties.putAll(createPropertiesFromConfig(config, COMMON_CONFIG_PREFIX));
+        properties.putAll(createPropertiesFromConfig(config, ADMIN_CLIENT_CONFIG_PREFIX));
+        properties.putAll(createPropertiesFromConfig(config, CONSUMER_CONFIG_PREFIX));
+        properties.putAll(createPropertiesFromConfig(config, PRODUCER_CONFIG_PREFIX));
+        return properties;
+    }
+
+    /**
+     * Gets the properties with the given prefix from the provided config.
+     *
+     * @param config The provided configuration.
+     * @param configPrefix The prefix of the properties to be returned.
+     * @return The properties with the prefix removed.
+     */
+    protected final Map<String, Object> createPropertiesFromConfig(final Config config, final String configPrefix) {
+        final Map<String, Object> properties = new HashMap<>();
```

Comment:
Yes, that should be changed. Although I don't think we can change the return type of the method that exposes the result as a bean: `Map<String, Object> createKafkaRuntimeConfig()`.

Reply:
Because we are "overwriting" a bean that is created by the Quarkus Kafka extension. While the injection uses a name, I assume that we cannot change the type, but I have not verified this yet. If your suggestion works, I am more than happy to change it.

Reply:
My understanding is that we do not even need the name anymore once we change the type ...

Reply:
You are right. So, I went a step further and create beans with our configuration classes instead of a map.